lib/Transforms/Scalar/ScalarReplAggregatesHLSL.cpp

```diff
@@ -3288,10 +3308,13 @@ static void ReplaceConstantWithInst(Constant *C, Value *V,
       Instruction *Inst = CE->getAsInstruction();
       Builder.Insert(Inst);
       Inst->replaceUsesOfWith(C, V);
-      ReplaceConstantWithInst(CE, Inst, Builder);
+      if (!ReplaceConstantWithInst(CE, Inst, Builder)) {
+        bReplacedAll = false;
+      }
```

Comment:
nit:
```suggestion
      if (!ReplaceConstantWithInst(CE, Inst, Builder))
        bReplacedAll = false;
```
see: https://llvm.org/docs/CodingStandards.html#don-t-use-braces-on-simple-single-statement-bodies-of-if-else-loop-statements